File: module_main/src/main/java/com/phone/module_main/model/MainModelImpl.java
package com.phone.module_main.model;

import com.phone.library_common.manager.RetrofitManager;
import com.phone.module_main.request.MainRequest;

import java.util.Map;

import io.reactivex.Observable;
import okhttp3.ResponseBody;

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2019/3/10 17:03
 * introduce :
 */

public class MainModelImpl implements IMainModel {

    private static final String TAG = "MainModelImpl";

    public MainModelImpl() {
    }

    @Override
    public Observable<ResponseBody> mainData(Map<String, String> bodyParams) {
        return RetrofitManager.instance().mRetrofit
                .create(MainRequest.class)
                .getMainData(bodyParams);
    }

}

File: module_main/src/main/java/com/phone/module_main/model/MainModelImpl.kt
package com.phone.module_main.model

import com.phone.library_network.manager.RetrofitManager.Companion.instance
import com.phone.module_main.request.MainRequest
import io.reactivex.Observable
import okhttp3.ResponseBody

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2020/3/10 17:10
 * introduce :
 */

class MainModelImpl : IMainModel {

    companion object {
        private val TAG = MainModelImpl::class.java.simpleName
    }

    override fun mainData(bodyParams: Map<String, String>): Observable<ResponseBody> {
        return instance().mRetrofit
            .create(MainRequest::class.java)
            .getMainData(bodyParams)
    }

}
